SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -54,3 +54,43 @@ extension Array {
         }
     }
 }
+
+extension CollectionType {
+    public var ip_middleValues: [Generator.Element]? {
```

Comment:
Looking forward, I see you're using this in mean, so maybe it has some value, but let's tweak the api a bit to return a non-optional array, and add some spacing / cleanup.  Mostly just saving count so it doesn't run multiple times, newlining array

```Swift
    func center() -> [Generator.Element] {
        guard !isEmpty else { return [] }
        let count = self.count
        let half = count / 2
        let center = left.advancedBy(1)
        if count.ip_isEven {
            let left = startIndex.advancedBy(half - 1)
            return [
                self[left],
                self[center]
            ]
        } else {
            return [
                self[center]
            ]
        }
    }
```